### DISCUSSION TITLE
OD-beta and LÖVE

### DISCUSSION THREAD

**Discussion by a-dekker** [eduardofilo/RG350_adam_image#253] [Ports]
OD-beta ships with the (fairly recent) LÖVE 11.3 2D Game Engine. There are some potentially nice love/lua games available, but I don't seem to be able to run any of them.
In the end I always and up with a black screen, while the game is running fine in the background.
Is LÖVE known to be broken on OD-beta, or is there a missing link I have not found yet?

P.s. testing and running a game is easy, it's just `love <gamezip.love>` or `love <love-gamedir>` from the terminal.

**Comment by eduardofilo** [comment]
Do you open the Terminal application in the console before launching the command?

**Reply by a-dekker** [reply]
yes, the usual procedure to stop gmenu2x and being able to open other apps.
Just to be sure I also created a test opk file, result is the same.

**Reply by eduardofilo** [reply]
OK.
I've tried a couple of games I have found and opening with the built-in interpreter in ODbeta, I get segmentation fault.
Then I have tried with the [lutro_libretro.so](https://docs.libretro.com/library/lutro/) core that a few weeks ago appeared in RetroArch and the error I get is:

```
/media/sdcard/roms/LOVE/bluerevolver_v051_demo.love:1: '=' expected near 'char(3)'
[ERROR] [Content]: Failed to load content
```

Same error in all the cases.

**Reply by a-dekker** [reply]
Good to see libretro is picking up support for this framework. Too bad it's not working for us yet.

A game like https://tangramgames.dk/games/duckmarines/ throws no errors for me, apart from the screen sound and input do work.
[Here](https://github.com/bbruno5/love-rg350/issues/1#issuecomment-678646478) is the mentioning of 11.3 not working on the "old" Opendingux due to the kernel. But thanks to OD-beta that should be solved.

Created a simple lua script to show a "hello world", love conf setting to fullscreen, but no sigar. I think I should forget about it for now.

**Reply by eduardofilo** [reply]
I have managed to get running with lutro_libretro some of the simple games here (some are a little slow):

https://github.com/libretro/libretro-lutro#sample-games

![IMG_20220529_155721](https://user-images.githubusercontent.com/2923728/170873678-eeed0c0a-bb88-4e00-87e1-d3af6cadfe78.jpg)

But they only work if after compressing with zip, I change the extension to `lutro`, not `love` although both are supposed to be supported by the core.

**Reply by SuperBadger77** [reply]
Does this mean we can use Lutro now? Or will it need a release?

**Reply by eduardofilo** [reply]
The Lutro core is not included in the last release of Adam, but it can be installed executing this installer:
https://github.com/eduardofilo/RG350_auto_ra_installer/releases/download/v2022-05-28/RA_ODBeta_Installer_v2022-05-28.opk

Although in a couple of weeks I plan to release new version of Adam with updated ODbeta and RetroArch.

**Reply by SuperBadger77** [reply]
> The Lutro core is not included in the last release of Adam, but it can be installed executing this installer: https://github.com/eduardofilo/RG350_auto_ra_installer/releases/download/v2022-05-28/RA_ODBeta_Installer_v2022-05-28.opk
> 
> Although in a couple of weeks I plan to release new version of Adam with updated ODbeta and RetroArch.

Cool - I'll await the new version :-)
